File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        nums = [v for v in nums if v > 0]

        for v in nums:
            idx = abs(v)-1
            if idx < len(nums) and nums[idx] > 0:
                nums[idx] *= -1


        for i in range(len(nums)):
            if nums[i] > 0:
                return i+1

        return len(nums) + 1

File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        const int n = 100002;
        int a[n] = {0};

        for(int val: nums){
            if (val > 0 && val < n){
                a[val] += 1;
            }
        }

        for (int i = 1; i < n; i++){
            if(!a[i]) return i;
        }

        return -1;
    }
};